python/src/iceberg/expressions/base.py

```diff
@@ -267,6 +226,68 @@ def __str__(self) -> str:
         return "false"
 
 
+class UnboundIn(BooleanExpression, UnboundPredicate):
+    """IN operation expression"""
+
+    def __init__(self, left: "UnboundReference", right: List[Literal]):  # pylint: disable=super-init-not-called
+        self._left = left
+        self._right = right
+
+    @property
+    def left(self) -> "UnboundReference":
+        return self._left
+
+    @property
+    def right(self) -> List[Literal]:
+        return self._right
+
+    def __eq__(self, other) -> bool:
+        return id(self) == id(other) or (isinstance(other, UnboundIn) and self.left == other.left and self.right == other.right)
+
+    def __invert__(self) -> Not:
+        return Not(self)
+
+    def __repr__(self) -> str:
```

Comment:
Does it make sense to use `@dataclass` so we get these for free?

Reply:
is the `__str__` representation key and should look like it does here - that would be one reason but I assume no

I also don't think dataclasses provide automatic encapsulation in any way like our obfuscation and use of properties does currently.

I'll make the switch and revert if needed.

Reply:
uses dataclasses